pandas/_libs/lib.pyx

```diff
@@ -2725,6 +2725,7 @@ def maybe_convert_objects(ndarray[object] objects,
                         # e.g. test_out_of_s_bounds_datetime64
                         seen.object_ = True
                         break
+                    break
```

Comment:
It appears we can remove the 4 `break`s here with:

```python
if convert_non_numeric:
    ...
else:
   ...
break
```
?

Reply:
Sure will update